packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -464,7 +536,9 @@ function isConditionalTest(node: TSESTree.Node): boolean {
 
   if (
     parent.type === AST_NODE_TYPES.ConditionalExpression &&
-    (parent.consequent === node || parent.alternate === node)
+    (parent.consequent === node ||
+      parent.alternate === node ||
+      node.type === AST_NODE_TYPES.UnaryExpression)
   ) {
```

Comment:
Oh, I'm realizing that these `UnaryExpression` recursions are unnecessary as well, I think, right?
```suggestion
    (parent.consequent === node || parent.alternate === node)
  ) {
```

Reply:
indeed, it was the second part of my "workaround", that your suggestion fixed.